moped-editor/src/views/projects/projectView/ProjectComponents/utils/makeComponentData.js

```diff
@@ -108,3 +108,61 @@ export const makeComponentInsertData = (projectId, component) => {
     feature_signals: { data: signalFeaturesToInsert },
   };
 };
+
+/**
+ * Assembles feature data based on I/O from the component attribute form. 
+ * It handles when a signal component changes to a non-signal component, 
+ * when a selected signal asset is cleared from the from the form input,
+ * or when the selected signal asset is changed to a different signal
+ * asset.
+ * @param {Object} signalFromForm - signal objected as returned by the signal
+ * autoomplete form option (which is essentially a signalr ecord from
+ *  socrata)
+ * @param {Object} clickedComponent  - the moped_project_component record that is
+ * currently being edited, including it's related feature data
+ * @returns {Object[]} signalsToCreate - an array of length 1 or 0 which optionally
+ * contains the signal feature record to be inserted
+ * @returns {Number[]} featureIdsToDelete - array of 0 or more feature record IDs
+ * which will be deleted
+ */
+export const getFeatureChangesFromComponentForm = (
```

Comment:
This all looks solid, but I found a small bug where a previously deleted signal features remains in state. To reproduce:
1. Create a signal component with a chosen signal from the signal dropdown
2. Click on the map so that no list items are expanded
3. Click on your signal component feature on the map to expand its associated list item
4. Edit attributes and clear the chosen signal from the form. Then, save
5. Click the red map edit button and you will see the previously deleted signal feature in yellow on the map

https://github.com/cityofaustin/atd-moped/assets/37249039/ad620ab4-8f35-4ff4-a13e-b5acdfdd60b0

Reply:
just tested this out, and it looks great. Having all of that logic in one place is the way! 🙌